### PR TITLE
Switch stdout/stderr to UTF-8 on Windows

### PR DESCRIPTION
**Pull request by easyaspi314**
Uses an adapted version of the fwprintf wrapper suggested by @t-mat,
and _setmode.

Also tries to work around classic MinGW not defining _O_U8TEXT.

### CODE REVIEW

**Comment by easyaspi314** [comment]
Awaiting Windows 7 verification from @sergeevabc.

File: xxhsum.c [hunk 1]
```diff
@@ -182,6 +182,50 @@ static FILE *XXH_fopen_wrapped(const char *filename, const wchar_t *mode)
     return f;
 }
 
+/*
+ * fprintf on Windows is, yet again, useless.
```
Comment (Cyan4973) [inline xxhsum.c]: Maybe not necessary preamble ...

Comment (easyaspi314) [reply xxhsum.c]: Fixed

**Comment by sergeevabc** [comment]
@easyaspi314, I’d gladly verify the binary if there is one.

**Comment by sergeevabc** [comment]
Ah, I see @easyaspi314 linked the binary [here][1]. Yes, it works as expected.

```
chcp 65001 
xxhsum.exe привет.txt
a2a5260369b4da80  привет.txt

chcp 1251
xxhsum.exe привет.txt
a2a5260369b4da80  привет.txt

chcp 866
xxhsum.exe привет.txt
a2a5260369b4da80  привет.txt
```

[1]: https://github.com/Cyan4973/xxHash/issues/293#issuecomment-594276980

**Pull request merged**